pkgs/circuits/halo2-secp256k1-wasm/src/eff_ecdsa.rs

```diff
@@ -1,46 +1,73 @@
 #![allow(non_snake_case)]
-use anyhow::{Context, Result};
+use anyhow::{Context, Ok, Result};
 use halo2_base::{
     gates::{circuit::builder::BaseCircuitBuilder, RangeChip},
-    halo2_proofs::halo2curves::{
-        bn256::Fr as Bn254Fr,
-        secp256k1::{Fp as Secp256k1Fp, Fq as Secp256k1Fq, Secp256k1Affine},
-    },
     utils::{BigPrimeField, CurveAffineExt},
-    Context as Halo2Context,
 };
 use halo2_ecc::{
     bigint::ProperCrtUint,
     ecc::{EcPoint, EccChip},
-    fields::{fp::FpChip, FieldChip},
-    secp256k1::{FpChip as Secp256k1FpChip, FqChip as Secp256k1FqChip},
+    fields::{fp, FieldChip},
 };
 use halo2_wasm::Halo2Wasm;
-use std::{cell::RefCell, rc::Rc};
-use wasm_bindgen::prelude::wasm_bindgen;
+use std::{cell::RefCell, marker::PhantomData, rc::Rc};
 
+use crate::consts::{FpChip, FqChip, CONTEXT_PHASE, F, FIXED_WINDOW_BITS, LIMB_BITS, NUM_LIMBS};
+
+// CF is the coordinate field of GA
+// SF is the scalar field of GA
 #[derive(Debug)]
-pub struct EffECDSAInputs {
-    s: Secp256k1Fq,
-    pk: Secp256k1Affine,
-    T: Secp256k1Affine,
-    U: Secp256k1Affine,
+pub struct EffECDSAInputs<CF, SF, GA>
+where
+    CF: BigPrimeField,
+    SF: BigPrimeField,
```

Comment:
now i am confused.
I thought you defined in `consts.rs`
```
pub type CF = secp256k1::Fp;
pub type SF = secp256k1::Fq;
```

why are CF and SF different types here?